src/hotspot/share/opto/type.cpp

```diff
@@ -5591,19 +5724,23 @@ const Type    *TypeInstKlassPtr::xmeet( const Type *t ) const {
     const TypeAryKlassPtr *tp = t->is_aryklassptr();
     int offset = meet_offset(tp->offset());
     PTR ptr = meet_ptr(tp->ptr());
+    InterfaceSet interfaces = meet_interfaces(tp);
+    InterfaceSet tp_interfaces = tp->_interfaces;
+    InterfaceSet this_interfaces = _interfaces;
 
     switch (ptr) {
     case TopPTR:
     case AnyNull:                // Fall 'down' to dual of object klass
       // For instances when a subclass meets a superclass we fall
       // below the centerline when the superclass is exact. We need to
       // do the same here.
-      if (klass()->equals(ciEnv::current()->Object_klass()) && !klass_is_exact()) {
+      if (klass()->equals(ciEnv::current()->Object_klass()) && this_interfaces.intersection_with(tp_interfaces).eq(this_interfaces) && !klass_is_exact()) {
```

Comment:
> this_interfaces.intersection_with(tp_interfaces).eq(this_interfaces)

Maybe a case for a helper method `InterfaceSet::contains(InterfaceSet)`?

Reply:
Indeed, there are several uses of this pattern. I will make that change.